### PR TITLE
FIX-2120 Move indexCurve first in LogCurveInfo

### LINKED ISSUE #2120: SpliceLogData - Ensure indexCurve is first in logCurveInfo
When splicing log data, there are some logs that does not return the indexCurve as the first element in the logCurveInfo list. This can lead to errors when creating the new spliced log.

Make sure to move the element that corresponds to indexCurve as the first element in the logCurveInfo that is used internally in SpliceLogDataWorker.

### PR DESCRIPTION
**Pull request by eliasbruvik**
[//]: # (Request Template Witsml Explorer)

[//]: # (Thank you for contributing to WITSML Explorer! Before submitting this PR, please fill in the following template.)

## Fixes

[//]: # (Write the GitHub issue number starting with #, or, for Equinor only, the Jira issue number starting with WE-)

This pull request fixes #2120 

## Description

[//]: # (Please include a written summary of the changes and which issue is fixed. List any dependencies that are required for this change._)

* Moved worker tools for log into a new `LogWorkerTools.cs` file.
* Changed the methods in the new file so that indexCurve is moved to the top of logCurveInfo.
* Added some tests

## Type of change

[//]: # (Mark any of the types of change that apply.)

* [ ] Bugfix
* [ ] New feature (non-breaking change which adds functionality)
* [x] Enhancement of existing functionality
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Impacted Areas in Application

[//]: # (List general components of the application that this PR will affect)

* [ ] Frontend
* [x] API
* [ ] WITSML
* [ ] Other (please describe)

## Checklist:

[//]: # (Please tick all the boxes or remove the ones that aren't needed and explain why)

*Communication*
* [ ] I have made corresponding changes to the documentation
* [ ] PR affects application security

*Code quality*
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] New code is covered by passing tests

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: Src/WitsmlExplorer.Api/Workers/LogWorkerTools.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+using System;
```
Comment (janmarius) [inline Src/WitsmlExplorer.Api/Workers/LogWorkerTools.cs]: Unused.

File: Src/WitsmlExplorer.Api/Workers/SpliceLogsWorker.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public override async Task<(WorkerResult, RefreshAction)> Execute(SpliceLogsJob
                     WitsmlLog logHeader = logHeaders.Logs.Find(l => l.Uid == logUid);
                     foreach (var mnemonic in logHeader.LogCurveInfo.Select(lci => lci.Mnemonic).Skip(1))
                     {
-                        WitsmlLogData logData = await WorkerTools.GetLogDataForCurve(GetTargetWitsmlClientOrThrow(), logHeader, mnemonic, Logger);
+                        WitsmlLogData logData = await LogWorkerTools.GetLogDataForCurve(GetTargetWitsmlClientOrThrow(), logHeader, mnemonic, Logger);
```
Comment (janmarius) [inline Src/WitsmlExplorer.Api/Workers/SpliceLogsWorker.cs:55]: The new changes make `using WitsmlExplorer.Api.Query;` unused.

File: Src/WitsmlExplorer.Api/Workers/WorkerTools.cs [hunk 3]
```diff
@@ -2,17 +2,12 @@
 using System.Linq;
```
Comment (janmarius) [inline Src/WitsmlExplorer.Api/Workers/WorkerTools.cs]: `using System.Collections.Generic;` unused.

**Review by janmarius — CHANGES_REQUESTED** [review]
There are a few unused imports in the code. Additionally, I suggest moving `WorkerTools.cs` and `LogWorkerTools.cs` into their own directory. Otherwise, it LGTM, good job! : )

File: Src/WitsmlExplorer.Api/Workers/LogWorkerTools.cs (hunk 1 above)
Comment (eliasbruvik) [reply Src/WitsmlExplorer.Api/Workers/LogWorkerTools.cs]: Good catch, removed the unused imports.

**Review by janmarius — APPROVED** [review]
*Reviewer checklist*
- [x] Read issue and PR
- [x] Pulled branch and manually tested
- [x] Verify that PR resolves issue
- [x] Reviewed the code

**Pull request merged**